torch/csrc/jit/codegen/cuda/runtime/block_welford_outer.cu

```diff
@@ -0,0 +1,180 @@
+namespace fused_reduction {
+namespace impl {
+
+// Scattered butterfly reduction
```

Comment:
It would be helpful to add more context here for the function implementation. Having a little trouble following the code.

Reply:
It looks like you're doing a reduction within each warps, which is why BDIMX must be a multiple of 32. Then you're doing a smem reduction across warps, but I'm uncertain how thread dimensions map clearly since this is for Outer.

Reply:
I'll write a more detailed description.